src/GafferScene/ResamplePrimitiveVariables.cpp

```diff
@@ -0,0 +1,119 @@
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "boost/format.hpp"
+#include "boost/array.hpp"
+
+#include "GafferScene/ResamplePrimitiveVariables.h"
+
+#include "IECore/MeshPrimitive.h"
+#include "IECore/CurvesPrimitive.h"
+#include "IECore/PointsPrimitive.h"
+#include "IECore/MeshAlgo.h"
+#include "IECore/CurvesAlgo.h"
+#include "IECore/PointsAlgo.h"
+
+using namespace IECore;
+using namespace Gaffer;
+using namespace GafferScene;
+
+IE_CORE_DEFINERUNTIMETYPED( ResamplePrimitiveVariables );
+
+size_t ResamplePrimitiveVariables::g_firstPlugIndex = 0;
+
+ResamplePrimitiveVariables::ResamplePrimitiveVariables( const std::string &name ) : PrimitiveVariableProcessor( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+
+	addChild( new IntPlug( "interpolation" ) );
+}
+
+ResamplePrimitiveVariables::~ResamplePrimitiveVariables()
+{
+}
+
+Gaffer::IntPlug *ResamplePrimitiveVariables::interpolationPlug()
+{
+	return getChild<IntPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::IntPlug *ResamplePrimitiveVariables::interpolationPlug() const
+{
+	return getChild<IntPlug>( g_firstPlugIndex );
+}
+
+void ResamplePrimitiveVariables::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	PrimitiveVariableProcessor::affects( input, outputs );
+
+	if( input == interpolationPlug() )
+	{
+		outputs.push_back( outPlug()->objectPlug() );
+	}
+}
+
+void ResamplePrimitiveVariables::hashProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	PrimitiveVariableProcessor::hashProcessedObject( path, context, h );
+
+	interpolationPlug()->hash( h );
+}
+
+void ResamplePrimitiveVariables::processPrimitiveVariable( const ScenePath &path, const Gaffer::Context *context, IECore::ConstPrimitivePtr inputGeometry, IECore::PrimitiveVariable &variable ) const
+{
+	static boost::array<PrimitiveVariable::Interpolation, 4> mapping = {
```

Comment:
Once you've added the min/max as suggested by Andrew, we can replace this with a simple cast since we won't need the range checking it provides.